tests/arc/topt_wasmoved_destroy_pairs.nim

```diff
@@ -6,93 +6,103 @@ discard """
 scope:
   def a: seq[seq[int]]
   def b: seq[seq[int]]
-  def x: seq[int] = f() (raises)
-  block L0:
-    scope:
-      if cond:
-        scope:
-          def _5: seq[int] = move x
-          add(name a, consume _5)
-          break L0
-    scope:
-      def _6: seq[int] = move x
-      add(name b, consume _6)
-  =destroy(name b)
-  =destroy(name a)
+  def x: seq[int] = f() -> [L0, Resume]
+  scope:
+    if cond:
+      scope:
+        def _5: seq[int] = move x
+        add(name a, consume _5)
+        goto [L2]
+  scope:
+    def _6: seq[int] = move x
+    add(name b, consume _6)
+  goto [L0, L3]
+  finally (L0):
+    =destroy(name b)
+    =destroy(name a)
+    continue {L3}
```

Comment:
This a minor regression compared to the previous destructor injection. Since the previous injection used data-flow analysis, it could prove that there was no unstructured scope exit (the `f()` call in line 9, in this case) where `a` or `b` were alive, so no `finally` was necessary.

At the moment, `mirgen` can only use a control-flow analysis, so a `finally` has to be used now. The performance impact is generally minimal, however, and future improvements to `mirgen` could allow the `finally` to omitted again.